skyportal/handlers/api/followup_request.py

```diff
@@ -429,6 +430,23 @@ def post_followup_request(
                 raise ValueError(
                     'Source has already been observed spectroscopically, not submitting request (as per constraint).'
                 )
+        if constraints.get("not_if_tns_classified", False):
+            # don't trigger if there is any source within 0.5 arcsec of the target
+            # that has a tns_name that contains "SN"
+            obj = session.scalars(
+                Obj.select(session.user_or_token).where(Obj.id == data['obj_id'])
+            ).first()
+            if obj is None:
+                raise ValueError(f'Could not find source with ID {data["obj_id"]}.')
+            stmt = Obj.select(session.user_or_token).where(
+                Obj.within(ca.Point(ra=obj.ra, dec=obj.dec), 0.5 / 3600),
+                Obj.tns_name.contains("SN"),
```

Comment:
Does this fail if the TNS name is something like, e.g., AT 2024asn?

Reply:
Nop, my understanding was that we only want to cancel triggering if its classified on TNS (has SN in its name), and otherwise we let it trigger.

@nabeelre is that correct? Or do you want the bot to not do anything as soon as there is an entry on TNS (even if not with SN)?

Reply:
this query just says "look for objects within 0.5 arcsec and with a TNS name with SN in it", if we find any results we cancel the trigger.

Reply:
Sorry, let me clarify. I was asking what the behavior would be if there's an unclassified source that just happens to have the letters "sn" in the randomized part of its name, for example, AT 2024asn. Unclassified but contain "sn." Is .contains() case sensitive?

Reply:
oh gotcha @nabeelre. It shouldn't be, but let me test that ASAP (haven't deployed yet anyway).

Reply:
@nabeelre answer is yes, it is case sensitive. Trying it locally, I can't trigger if its SN2023ab, but I can if it's sn2023ab. Maybe we want it to not be case sensitive but only check if the string starts with SN. Might make more sense in case people add tns name with an endpoint without capitalizing that

Reply:
But as is, this works

Reply:
Opening a PR now to replace `contains` by `startswith` while keeping it case sensitive, seems like the safest way to go

Reply:
`sa.func.lower(Obj.tns_name).startswith("sn")`